com.reprezen.swagedit/src/com/reprezen/swagedit/json/JsonDocumentManager.java

```diff
@@ -90,6 +94,34 @@ public JsonNode getDocument(URL url) {
         return document;
     }
 
+    /*
+     * This method tries to read the content of the URL as a JSON or YAML document depending on the configuration of the
+     * object mapper passed as parameter.
+     * 
+     * It also forces the content of the URL to be read with UTF-8 encoding, considering that only UTF encodings are
+     * valid encodings for JSON inputs.
+     */
+    protected JsonNode doRead(ObjectMapper mapper, URL url) throws Exception {
+        try (InputStream stream = url.openStream()) {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(stream, "UTF-8"))) {
+                return mapper.readTree(reader);
+            } catch (CharConversionException e) {
+                throw new Exception(e);
+            }
+        } catch (IOException e) {
+            throw new Exception(e);
```

Comment:
I wonder why we need to wrap existing `IOException ` and `CharConversionException ` into a generic `Exception`...